File: src/main/java/com/github/hhhzzzsss/songplayer/song/Instrument.java
package com.github.hhhzzzsss.songplayer.song;

public enum Instrument {
	HARP(54),
	BASEDRUM(0),
	SNARE(0),
	HAT(0),
	BASS(30),
	FLUTE(66),
	BELL(78),
	GUITAR(42),
	CHIME(78),
	XYLOPHONE(78),
	IRON_XYLOPHONE(54),
	COW_BELL(66),
	DIDGERIDOO(30),
	BIT(54),
	BANJO(54),
	PLING(54);

	public final String instrumentName;
	public final int instrumentId;
	public final int midiOffset;

	Instrument(int offset) {
		this.instrumentName = name().toLowerCase();
		this.instrumentId = ordinal();
		this.midiOffset = offset;
	}

	private static final Instrument[] values = values();
	public static Instrument getInstrumentFromId(int instrumentId) {
		return values[instrumentId];
	}
}

File: src/main/java/com/github/hhhzzzsss/songplayer/song/Instrument.kt
package com.github.hhhzzzsss.songplayer.song

import java.util.*

enum class Instrument(offset: Int) {
    HARP(54),
    BASEDRUM(0),
    SNARE(0),
    HAT(0),
    BASS(30),
    FLUTE(66),
    BELL(78),
    GUITAR(42),
    CHIME(78),
    XYLOPHONE(78),
    IRON_XYLOPHONE(54),
    COW_BELL(66),
    DIDGERIDOO(30),
    BIT(54),
    BANJO(54),
    PLING(54);

    @JvmField
	val instrumentName: String = name.lowercase(Locale.getDefault())
    @JvmField
	val instrumentId: Int = ordinal
    @JvmField
	val midiOffset: Int = offset

    companion object {
        @JvmStatic
		fun getInstrumentFromId(instrumentId: Int): Instrument {
            return entries[instrumentId]
        }
    }
}